spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -128,6 +130,7 @@ public String deploy(AppDeploymentRequest request) {
 				if (useDynamicPort) {
 					args.put(SERVER_PORT_KEY, String.valueOf(port));
 				}
+				args.put("spring.cloud.stream.metrics.key", deploymentId + ".${spring.application.index}");
```

Comment:
do you want to hardcode port here is use `spring.cloud.application.guid` what I think this ref was supposed to be?